app/src/parser_impl.c

```
@@ -180,12 +220,207 @@ parser_error_t parser_getValidClist(uint16_t *clist_token_index, uint16_t *num_a
 }
 
 bool items_isNullField(uint16_t json_token_index) {
-    parsed_json_t *json_all = &(parser_getParserTxObj()->json);
+    parsed_json_t *json_all = &(parser_getParserJsonObj()->json);
     jsmntok_t *token = &(json_all->tokens[json_token_index]);
 
     if (token->end - token->start != sizeof("null") - 1) return false;
 
-    return (MEMCMP("null", json_all->buffer + token->start, token->end - token->start) == 0);
+    return CMP_STRING_AND_BUFFER("null", json_all->buffer + token->start, token->end - token->start);
+}
+
+parser_error_t parser_createJsonTemplate(parser_context_t *ctx) {
+    uint8_t tx_type = 0;
+    char address[65] = {0};
+    uint16_t address_len = 0;
+    chunk_t chunks[MAX_FIELDS_IN_INPUT_DATA] = {0};
+
+    parser_readSingleByte(ctx, &tx_type);
+
+    for (int i = 0; i < MAX_FIELDS_IN_INPUT_DATA; i++) {
+        parser_readSingleByte(ctx, &chunks[i].len);
+        if (chunks[i].len > 0) {
+            parser_readBytes(ctx, (uint8_t **)&chunks[i].data, chunks[i].len);
```

Can we add checking for the return of these functions., so we can propagate the error.